File: backend/src/Domain/Entities/Produto.cs
namespace StockManager.Domain.Entities;

public class Produto
{
    public int Id { get; set; }
    public string Nome { get; set; } = string.Empty;
    public string Descricao { get; set; } = string.Empty;
    public string SKU { get; set; } = string.Empty;
    public decimal Preco { get; set; }
    public int QuantidadeEmEstoque { get; set; }
    public int QuantidadeReservada { get; set; }
    public DateTime CriadoEm { get; set; } = DateTime.UtcNow;
    public DateTime AtualizadoEm { get; set; } = DateTime.UtcNow;

    public ICollection<MovimentacaoEstoque> Movimentacoes { get; set; } = new List<MovimentacaoEstoque>();

    public int QuantidadeDisponivel => QuantidadeEmEstoque - QuantidadeReservada;
}

File: frontend/src/app/services/produto.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { environment } from '../../environments/environment';

export interface Produto {
  id?: number;
  nome: string;
  descricao: string;
  sku: string;
  preco: number;
  quantidadeEmEstoque?: number;
  quantidadeReservada?: number;
  quantidadeDisponivel?: number;
}

@Injectable({ providedIn: 'root' })
export class ProdutoService {
  private apiUrl = `${environment.apiUrl}/produtos`;

  constructor(private http: HttpClient) {}

  getAll() { return this.http.get<Produto[]>(this.apiUrl); }
  getById(id: number) { return this.http.get<Produto>(`${this.apiUrl}/${id}`); }
  create(produto: Produto) { return this.http.post<Produto>(this.apiUrl, produto); }
  update(id: number, produto: Produto) { return this.http.put(`${this.apiUrl}/${id}`, produto); }
  delete(id: number) { return this.http.delete(`${this.apiUrl}/${id}`); }
}